// react-rainbow-components/components/ColorInput/styled.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyledColorSample = exports.StyledAlpha = exports.StyledAlphaInput = exports.StyledInput = exports.ColorInputContainer = exports.StyledContent = exports.StyledCard = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _attachThemeAttrs = _interopRequireDefault(require("../../styles/helpers/attachThemeAttrs"));

var _Card = _interopRequireDefault(require("../Card"));

var _styled = require("../PhoneInput/styled");

function _templateObject7() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    width: 1.5rem;\n    height: 1.5rem;\n    border-radius: 50%;\n    background-color: ", ";\n    border: solid 1px\n        ", ";\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    position: relative;\n    display: flex;\n    align-items: center;\n    color: ", ";\n    font-weight: bold;\n\n    :before {\n        content: '';\n        position: absolute\n        height: 1.5rem;\n        margin: 0 auto;\n\n        border-left: 1px solid ", ";\n        ", ";\n\n        ", ";\n    }\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    width: 2.5em;\n    text-align: center;\n    font-weight: normal;\n    padding-left: 8px;\n\n    :focus,\n    :active {\n        outline: 0;\n        padding-left: 8px;\n    }\n\n    &::-moz-focus-inner {\n        border: 0;\n        padding-left: 8px;\n    }\n\n    /* Chrome, Safari, Edge, Opera */\n    &::-webkit-outer-spin-button,\n    &::-webkit-inner-spin-button {\n        -webkit-appearance: none;\n        margin: 0;\n    }\n\n    /* Firefox */\n    &[type='number'] {\n        -moz-appearance: textfield;\n    }\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    display: flex;\n    padding-left: 1.2rem;\n\n    ::-moz-focus-inner {\n        border: 0;\n        padding-left: 1.2rem;\n    }\n\n    :focus,\n    :active {\n        outline: 0;\n        padding-left: 1.2rem;\n    }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    display: flex;\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    width: 19rem;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    padding: 10px;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var StyledCard = (0, _styledComponents["default"])(_Card["default"])(_templateObject());
exports.StyledCard = StyledCard;

var StyledContent = _styledComponents["default"].div(_templateObject2());

exports.StyledContent = StyledContent;

var ColorInputContainer = _styledComponents["default"].div(_templateObject3());

exports.ColorInputContainer = ColorInputContainer;
var StyledInput = (0, _styledComponents["default"])(_styled.StyledInput)(_templateObject4());
exports.StyledInput = StyledInput;
var StyledAlphaInput = (0, _styledComponents["default"])(_styled.StyledInput)(_templateObject5());
exports.StyledAlphaInput = StyledAlphaInput;
var StyledAlpha = (0, _attachThemeAttrs["default"])(_styledComponents["default"].div)(_templateObject6(), function (props) {
  return props.palette.text.header;
}, function (props) {
  return props.palette.border.divider;
}, function (props) {
  return props.readOnly && "\n            border-left: 1px solid transparent;\n            padding-left: 8px;\n        ";
}, function (props) {
  return props.disabled && "\n            border-left: 1px solid ".concat(props.palette.border.disabled, ";\n            }\n        ");
});
exports.StyledAlpha = StyledAlpha;
var StyledColorSample = (0, _attachThemeAttrs["default"])(_styledComponents["default"].div)(_templateObject7(), function (props) {
  return props.backgroundColor;
}, function (props) {
  return props.palette.border.disabled;
});
exports.StyledColorSample = StyledColorSample;
